src/provider/CoinbaseWalletProvider.ts

```diff
@@ -235,15 +235,17 @@ export class CoinbaseWalletProvider
     address: string,
     symbol?: string,
     decimals?: number,
-    image?: string
+    image?: string,
+    chainId?: number
   ): Promise<boolean> {
     const relay = await this.initializeRelay()
     const result = await relay.watchAsset(
       type,
       address,
       symbol,
       decimals,
-      image
+      image,
+      chainId?.toString()
```

Comment:
```suggestion
      chainid ?? String(chainid)
```
since `.toString()` doesn't guarantee a string return value, but `String()` does

Reply:
Not sure if we should, we don't want to guarantee a string return value. In cases where it doesn't find a chainId, we just shouldn't pass any string, that's why it is an optional param. To pass "null" or "undefined" (as strings) wouldn't help in this case since it wouldn't refer to any network.

Reply:
How

Reply:
The issue is that any object can have a `toString` method that doesn't return a string.

With my suggestion, it will return `null` or `undefined` if those are what's passed, otherwise, it will *guarantee* a string return.

Reply:
not sure what we're looking for on the frontend, but if `undefined` is passed in `chainid ?? String(chainid)` would return the string `'undefined'`. not sure if i'm following the point about any object being passed in here - wouldn't it just be a number?

Reply:
oh right, sorry - the suggestion should be:
```suggestion
      chainId == null ? chainId : String(chainId)
```

This is JS; any kind of value could be passed in.

Reply:
> not sure what we're looking for on the frontend, but if `undefined` is passed in `chainid ?? String(chainid)` would return the string `'undefined'`. not sure if i'm following the point about any object being passed in here - wouldn't it just be a number?

@jeongmincho Actually it wouldn't happen because if chainId is a falsy value, it would return undefined/null and never process `String(chainId)`




> The issue is that any object can have a `toString` method that doesn't return a string.
> 
> With my suggestion, it will return `null` or `undefined` if those are what's passed, otherwise, it will _guarantee_ a string return.

@ljharb, Isn't it already guaranteed here https://github.com/coinbase/coinbase-wallet-sdk/blob/9c58ef59625d0dba7d5c151ea984815fe8c61259/src/provider/CoinbaseWalletProvider.ts#L854-L858?

Reply:
no - `chainId` is a prop passed in by a user. a prop can be literally any value; TS doesn't exist at runtime.

What you linked guarantees that you only *pass* an integer number - but users can pass anything they want.

Reply:
@pedro-rosario `??` returns the right-hand side if left-hand side is undefined/null

Reply:
Oh sry, you're right